lizard_auth_server/views_api.py

```diff
@@ -296,7 +296,72 @@ def form_invalid(self, form):
 
     def get_organisations(self, portal):
         return {
-            'success': True,
             'organisations': [
                 organisation.as_dict()
                 for organisation in models.Organisation.objects.all()]}
+
+class RolesView(FormView):
+    """
+    View that can be used to respond with serialized Roles.
+    """
+    form_class = forms.DecryptForm
+
+    @method_decorator(csrf_exempt)
+    def dispatch(self, request, *args, **kwargs):
+        return super(RolesView, self).dispatch(
+            request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        return super(RolesView, self).post(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        return JsonResponse(self.get_roles(form.portal))
+
+    def form_invalid(self, form):
+        logger.error(
+            'Error while decrypting roles form: {}'.format(form.errors.as_text())
+        )
+        return HttpResponseBadRequest('Bad signature')
+
+    def get_roles(self, portal):
+        return {"roles": [role.as_dict()
+                          for role in models.Role.objects.all()
```

Comment:
objects.filter(portal=portal) en dan de if weg, dan gebeurt het in SQL ipv Python